galite-core/src/main/kotlin/org/kopi/galite/report/Triggers.kt

```diff
@@ -15,6 +15,244 @@
+        var result = 0
+        for (i in 0 until childCount) {
+          val child = row.getChildAt(i) as VReportRow
+          val value = child.getValueAt(column) as Int?
+          if (value != null) {
+            result += value
+          }
+        }
+        return result
+      }
+    }
+  }
+
+  /**
+   * Compute the number of entries in a report column
+   */
+  fun countInteger(c: VReportColumn): VCalculateColumn {
+    return object : VCCDepthFirstCircuitN() {
+      override fun evalNode(row: VReportRow, column: Int): Any {
+        val childCount = row.childCount
+        return if (row.level > 1) {
+          // the value of a node is the sum of
+          // other nodes (contain no leafs)
+          var result = 0
+          for (i in 0 until childCount) {
+            val child = row.getChildAt(i) as VReportRow
+            val value = child.getValueAt(column) as Int?
+            if (value != null) {
+              result += value
+            }
+          }
+          result
+        } else {
+          // the value is the number of the childs
+          // if the childs are leafs
+          childCount
+        }
+      }
+    }
+  }
+
+  /**
+   * Compute the fixed sum in a report column
+   */
+  fun sumFixed(c: VReportColumn): VCalculateColumn {
+    return object : VCCDepthFirstCircuitN() {
+      override fun evalNode(row: VReportRow, column: Int): Any {
+        val childCount = row.childCount
+        var result = NotNullFixed(0, 2)
+        for (i in 0 until childCount) {
+          val child = row.getChildAt(i) as VReportRow
+          val value = child.getValueAt(column) as NotNullFixed?
+          if (value != null) {
+            result = result.add(value)
+          }
+        }
+        return result
+      }
+    }
+  }
+
+  /**
+   * Compute the integer sum in a report column
+   */
+  fun sumNullInteger(c: VReportColumn): VCalculateColumn {
+    return object : VCCDepthFirstCircuitN() {
+      override fun evalNode(row: VReportRow, column: Int): Any? {
+        val childCount = row.childCount
+        var result = 0
+        var valueFound = false
+        for (i in 0 until childCount) {
+          val child = row.getChildAt(i) as VReportRow
+          val value = child.getValueAt(column) as Int?
+          if (value != null) {
+            valueFound = true
+            result += value
+          }
+        }
+        return if (valueFound) result else null
+      }
+    }
+  }
+
+  /**
+   * Compute the fixed sum in a report column
+   */
+  fun sumNullFixed(c: VReportColumn): VCalculateColumn {
+    return object : VCCDepthFirstCircuitN() {
+      override fun evalNode(row: VReportRow, column: Int): Any? {
+        val childCount = row.childCount
+        var valueFound = false
+        var result = NotNullFixed(0, 2)
+        for (i in 0 until childCount) {
+          val child = row.getChildAt(i) as VReportRow
+          val value = child.getValueAt(column) as NotNullFixed?
```

Comment:
can you check this value because its notNullFixed so it can't be null

Reply:
I suggest to keep it as it is for now